scripts/install.sh

```diff
@@ -4,10 +4,39 @@ set -e
 
 os=$(uname | tr '[:upper:]' '[:lower:]')
 arch=$(uname -m | tr '[:upper:]' '[:lower:]' | sed -e s/x86_64/amd64/)
+if [ "$arch" = "aarch64" ]; then
+  arch="arm64"
+fi
+
+url="https://infracost.io/downloads/latest"
+tar="infracost-$os-$arch.tar.gz"
 echo "Downloading latest release of infracost-$os-$arch..."
-curl -sL https://github.com/infracost/infracost/releases/latest/download/infracost-$os-$arch.tar.gz | tar xz -C /tmp
+curl -sL $url/$tar -o /tmp/$tar
+echo
+
+code=$(curl -s -L -o /dev/null -w "%{http_code}" $url/$tar.sha256)
+if [ "$code" = "404" ]; then
+  echo "Checksum for infracost-$os-$arch release not found, skipping checksum validation"
+else
+  if [ -x "$(command -v shasum)" ]; then
+    echo "Validating checksum for infracost-$os-$arch..."
+    curl -sL $url/$tar.sha256 -o /tmp/$tar.sha256
+    (cd /tmp/;shasum -sqc $tar.sha256)
+    rm /tmp/$tar.sha256
+  else
+    echo "The shasum command could not be found, skipping checksum validation"
```

Comment:
I'm wondering if we should avoid logging here and on line 19 if there's a 404. In both cases, there's nothing the user can do (since the install proceeds anyway), so why warn them? The one we want to flag-up is when the checksum fails. What does that log look like? I'd even consider not logging if the checksum passes.

I say this based on the `brew install` experience, it's so painless/unalarming that I didn't even know it was checking the checksum.

Reply:
I don't have a strong opinion of this - however, I remember seeing a brew issue (can't find it at the moment) where someone mentioned they wanted to have a warning if the cask had disabled `sha256`. Which suggests that we might want to be as verbose as possible. 

The warning message when checksum fails is as follows:

```
infracost-darwin-amd64.tar.gz: FAILED
shasum: WARNING: 1 computed checksum did NOT match
```

This is directly from the shasum command. We could capture output and modify message if you want.

Reply:
I'd optimize this flow for getting to success as soon as possible. The 404 log doesn't have any actionable things for the user to do, so I'd probably skip that so we can merge this PR. The missing shasum command gives me a bit of pause - maybe "Skipping checksum validation as the shasum command could not be found, no action needed." is clearer that we don't want the user to do anything, this is just FYI.

Re the failure log, maybe it could say "Installation checksum failed! This could be a security issue so please email hello@infracost.io for help."
For now, I guess it's ok to continue with the installation even if the checksum failed until we revisit this side of things next month.